libtsuba/src/RDG.cpp

```diff
@@ -109,6 +116,81 @@ MasterPropName(unsigned i) {
   return std::string(kMasterNodesPropName) + "_" + std::to_string(i);
 }
 
+katana::Result<std::shared_ptr<arrow::ChunkedArray>>
+LargeStringToChunkedString(
+    const std::shared_ptr<arrow::LargeStringArray>& arr) {
+  std::vector<std::shared_ptr<arrow::Array>> chunks;
+
+  arrow::StringBuilder builder;
+
+  uint64_t inserted = 0;
+  constexpr uint64_t kMaxSize = 2147483646;
+
```

Comment:
0x7ffffffe?

Reply:
Oh, that is the max size of a not large string array?

Reply:
Yeah, taken straight out of an arrow error message 😄